alerter.cpp

```diff
@@ -5,27 +5,31 @@ int alertFailureCount = 0;
 
 int networkAlertStub(float celcius) {
     std::cout << "ALERT: Temperature is " << celcius << " celcius.\n";
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
+   if(celcius>200)
+        return 500;
+    else      
+        return 200;   
+}
+
+float convertFarenehitToCelcius(float farenheit)
+{
+    float celcius;
+    celcius = ((farenheit-32)*5/9);
+    return celcius;
 }
 
 void alertInCelcius(float farenheit) {
-    float celcius = (farenheit - 32) * 5 / 9;
+    float celcius = convertFarenehitToCelcius (farenheit);
     int returnCode = networkAlertStub(celcius);

```

Comment:
Name with stub is used for test code. If it is a test stub then function pointer could be used to segregate test stub code from production code. Do not mix test code with production code.